ui/desktop/app/styles/app.scss

```diff
@@ -530,3 +530,111 @@ $chevron-down: url("data:image/svg+xml;utf-8,<svg viewBox='0 0 24 24' fill='%231
     background-color: var(--token-color-palette-neutral-50);
   }
 }
+
+.details-screen-body {
+  .rose-layout-body-content {
+    margin-top: 1.5rem;
+  }
+}
+
+// Session Details - Credentials Panel
+.credential-panel-header {
+  display: flex;
+  justify-content: space-between;
+  align-items: flex-end;
+  margin-bottom: 0.5rem;
+
+  .title {
+    font-weight: var(--token-typography-font-weight-bold);
+  }
+
+  /* stylelint-disable selector-class-pattern */
+  .hds-form-field--layout-flag {
+    margin-bottom: 0;
+  }
+}
+
+.credential-panel-body {
+  border: 0.063rem solid var(--token-color-border-primary);
+  border-radius: 0.313rem;
+  display: flex;
+  margin-top: 0.75rem;
+
+  .credential-header {
+    padding: 1.5rem;
+    width: 33%;
+  }
+}
+
+.credential-secret {
+  width: 66%;
+  margin: 0.75rem;
+
+  .secret-container {
+    margin: 0.75rem 0.5rem;
+    border-bottom: 0.0625rem solid var(--token-color-border-primary);
+
+    &:last-child {
+      border-bottom: 0;
+    }
+
+    .secret-key {
+      font-weight: var(--token-typography-font-weight-semibold);
+    }
+
+    .hidden-secret {
+      margin: 0.25rem 0;
+
+      .rose-button-ghost {
+        margin-bottom: 0;
+        color: var(--ui-gray-subtler-2);
+      }
+    }
+  }
+}
+
+.credential-name {
+  font-weight: var(--token-typography-font-weight-semibold);
+}
+
+.raw-credential-panel-body {
+  $border-width: 0.063rem;
+
+  border: 0.063rem solid var(--token-color-border-primary);
+  border-radius: 0.313rem;
+  display: block;
+  margin-top: 0.75rem;
+
+  .raw-credential-header {
+    padding: 1.5rem 1.5rem 0.75rem;
+  }
+
+  .raw-credential-title {
+    display: flex;
+    justify-content: space-between;
+
+    .hds-copy-button {
+      padding: 0;
+      background: unset;
+      border: unset;
+      box-shadow: unset;
+      min-height: unset;
+
+      .hds-button__icon {
+        color: var(--ui-gray-subtler-2);
+      }
+    }
+  }
+
+  .secret-raw .rose-codemirror-container {
+    position: relative;
+    right: $border-width;
+    top: $border-width;
+    width: calc(100% + (2 * $border-width));
+  }
+
+  .secret-raw .CodeMirror {
+    border-bottom-left-radius: 0.313rem;
+    border-bottom-right-radius: 0.313rem;
+  }
+}
```

Comment:
What are these needed for?

Reply:
The container styles are to make the editor the width of the parent plus the border width to match the wireframe. Happy to update this with a better solution.

The `border-bottom-[left|right]-radius` is to add radius to just the bottom corners. 
<img width="713" alt="Screen Shot 2023-08-23 at 8 49 46 AM" src="https://github.com/hashicorp/boundary-ui/assets/29386339/b21985a9-31bc-45d2-862c-6151d89baa81">

Reply:
If the issue is the size is not correct, should we be instead passing the correct sizing info to code mirror itself rather than trying to override it ourselves?

Reply:
It isn't that the size isn't correct. Just that sometimes we need the border-radius and in some cases (workers) we don't use a border-radius. As for the width, this is the only place I have found that the width defined by code mirror was not sufficient. We could add border-radius to code mirror by default but I would still need to remove the top left and right for my work. I would prefer that change happen outside of this PR though.

Reply:
It seems like to me you're trying to compensate for the extra border that's on the container that includes the header and the editor.

If you moved the border to just the header instead, do you need to fiddle with the code editor position and calculate width at all?

Reply:
Absolutely doable. Fixed now. Thanks for that suggestion.